packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -174,8 +229,16 @@ function GridFilterForm(props: GridFilterFormProps) {
     <GridFilterFormRoot className={classes.root}>
       <rootProps.components.BaseFormControl
         variant="standard"
-        sx={{ flexShrink: 0, justifyContent: 'flex-end', marginRight: 0.5, marginBottom: 0.2 }}
-        {...rootProps.componentsProps?.baseFormControl}
+        {...baseFormControlProps}
+        {...deleteIconProps}
+        sx={{
+          flexShrink: 0,
+          justifyContent: 'flex-end',
+          marginRight: 0.5,
+          marginBottom: 0.2,
+          ...(deleteIconProps?.sx || {}),
+        }}
+        className={clsx(classes.closeIcon, baseFormControlProps.className)}
```

Comment:
A custom class name will be overwritten if not applied individually.

```suggestion
        className={clsx(classes.closeIcon, baseFormControlProps.className, deleteIconProps.className)}
```

x other components.